### ISSUE TITLE
Unable to change graphqlPath

### ISSUE THREAD

**Issue by josephharveyangeles** [apollographql/apollo-server#1617]
Using the apollo-server@2.0.6, I can change the port but I'm having trouble changing the graphql path. I tried the following to no avail.

```javascript
const server = new ApolloServer({...});
server.setGraphQLPath('/graphql');
server.graphqlPath = '/graphql';
server.listen({port: 3000, path: '/graphql'}).then(({ url }) => {
   console.log(`🚀 Server ready at ${url}`);
});
```
logs: `🚀 Server ready at http://localhost:3000`

**Comment by sbrichardson** [comment]
I'm assuming you are using the basic example with the `apollo-server` package?

#### tl;dr:
To set a custom path, you would need to use a middleware option such as `apollo-server-express`

#### Long Explanation:

The basic example from the docs creates a server that listens for requests on all paths, on either the default port `4000`, or a port you define in the listen step: `server.listen({ port: 3000 })... `

The reason the path option is ignored is the basic example isn't doing any routing or applying any middleware to the requests. It's only using basic logic internally to determine if you sent a valid graphql request, or requested the playground etc. You can confirm this a few ways:

With a default config, you can startup the server and use a browser to visit `http://localhost:4000/any/random/path`.  You should get the Playground UI.

You can also disable the Playground by passing `playground: false` when you instantiate the ApolloServer. Access via the browser,  you will get the basic graphql GET request error. Or send a valid GraphQL request to `http://localhost:3000/any/random/path` and it should still work.

Hopefully that helps? This question has come up a few times, so I wanted to provide a better explanation.

**Comment by josephharveyangeles** [comment]
@sbrichardson thanks for the clarification. Does it mean that the `apollo-server` was not meant to use for use cases where you need to change the path? What is the advantage of `apollo-server` over `apollo-server-express`? Is `apollo-server` for example purposes only and we should opt to use the middleware (`apollo-server-express`)? Sorry, it's quite confusing for newcomers like me.

**Comment by sbrichardson** [comment]
using apollo-server-express is technically the same thing as using the vanilla apollo-server, you just gain the ability to add middleware and other features. 

When you use the vanilla/basic apollo-graphql, it's actually using express in the backend. I'm not an Apollo employee so can't comment on it directly, just wanted to clarify!

**Comment by josephharveyangeles** [comment]
thanks @sbrichardson! Thanks for all the information, that clears it up for me.
